### PR TITLE
feat: add traversal paths

### PR DESCRIPTION
**Pull request by numb3r3**

### CODE REVIEW

File: server/clip_server/executors/clip_onnx.py [hunk 1]
```diff
@@ -59,7 +61,7 @@ def __init__(
             and hasattr(self.runtime_args, 'replicas')
         ):
             replicas = getattr(self.runtime_args, 'replicas', 1)
-            num_threads = max(1, torch.get_num_threads() // replicas)
+            num_threads = max(1, torch.get_num_threads() * 2 // replicas)
```
Comment (ZiniuYu) [inline server/clip_server/executors/clip_onnx.py:64]: why *2

Comment (numb3r3) [reply server/clip_server/executors/clip_onnx.py:64]: 50% CPU utilization with * 1

**Review by ZiniuYu — COMMENTED** [review]
Need to add some test cases

File: client/clip_client/client.py [hunk 2]
```diff
@@ -184,12 +187,18 @@ def _iter_doc(self, content) -> Generator['Document', None, None]:
...
+
+        payload = dict(
             on='/',
             inputs=self._iter_doc(content),
-            request_size=kwargs.get('batch_size', 8),
+            request_size=kwargs.get('batch_size', 32),
+            parameters=parameters,
             total_docs=len(content) if hasattr(content, '__len__') else None,
         )
+        return payload
```
Comment (ZiniuYu) [inline client/clip_client/client.py:203]: combine these lines

**Review by ZiniuYu — CHANGES_REQUESTED** [review]

**Review by ZiniuYu — APPROVED** [review]
LGTM

**Comment by hanxiao** [comment]
traversal path should not be handled in CAS, user should first traverse and then send via client

File: client/clip_client/client.py [hunk 3]
```diff
@@ -158,17 +158,13 @@ def _iter_doc(self, content) -> Generator['Document', None, None]:
...
             elif isinstance(c, Document):
+                self._return_plain = False
                 if c.content_type in ('text', 'blob'):
-                    self._return_plain = False
                     yield c
                 elif not c.blob and c.uri:
                     c.load_uri_to_blob()
-                    self._return_plain = False
-                    yield c
-                elif c.tensor is not None:
```
Comment (ZiniuYu) [inline client/clip_client/client.py:168]: should keep this line

**Comment by numb3r3** [comment]
> traversal path should not be handled in CAS, user should first traverse and then send via client

The executors in the CAS will then be uploaded to the hub. So I think the `traversal_paths` support is necessary.  And of course, if the user uses CAS client, the recommended way is to traverse first (the `traversal_paths` will not be mentioned in CAS doc).

**Pull request merged**